gridded_etl_tools/utils/extractor.py

```diff
@@ -16,92 +16,86 @@
+    batch
+        A 2D list of lists (or any sequential type) containing arguments to be forwarded to the given batch
+        processor
+
+    Returns
+    -------
+    bool
+        `True` if any of the batch calls returns `True`, `False` otherwise
+    """
+    # Success remains Trueuntil the first unsuccessful request is completed
+    success = True
 
-    def __init__(self, dm: dataset_manager.DatasetManager):
-        """
-        Create a new Extractor object by associating a Dataset Manager with it.
-
-        Parameters
-        ----------
-        dm
-            Source data for this dataset manager will be extracted
-        """
-        self.dm = dm
-
-    def pool(self, batch_processor: typing.Callable[..., bool], batch: typing.Sequence[typing.Sequence]) -> bool:
-        """
-        Launch a batch of requests simultaneously, wait for them all to complete, then return a boolean indicating
-        whether any of the requests were successful.
+    # Time the download
+    start_downloading = time.time()
 
-        The batch processor is any generic callable that accepts the arguments provided in each entry of the batch.
-        Each entry in the batch is a list of arguments to be forwarded to the batch processor. The batch is submitted
-        to a thread pool which will call the batch processor on every entry of the batch, launching all the calls in
-        parallel and waiting for them to complete.
+    # Thread count is one less than the amount of CPU available, unless there is only 1 CPU
+    thread_count = max(1, multiprocessing.cpu_count() - 1)
 
-        If any of the calls returns `True`, this will return `True`, otherwise it will return `False`.
+    # run downloads
+    if len(batch) > 0:
+        log.info(f"Submitting request for {len(batch)} datasets in parallel using {thread_count} threads.")
 
-        Parameters
-        ----------
-        batch_processor
-            Function which will be called repeatedly, being passed a new entry in the batch each time it's called
-        batch
-            A 2D list of lists (or any sequential type) containing arguments to be forwarded to the given batch
-            processor
+        # download in parallel
+        with multiprocessing.pool.ThreadPool(processes=thread_count) as pool:
+            for result in pool.starmap(batch_processor, batch):
+                if not result:
+                    success = False
 
-        Returns
-        -------
-        bool
-            `True` if any of the batch calls returns `True`, `False` otherwise
-        """
-        # Success remains false until the first successful request is completed
+        log.info(f"Downloading took {(time.time() - start_downloading) / 60:.2f} minutes")
+    else:
+        log.info("No requests prepared to be submitted, please check batch_processor is functioning properly.")
         success = False
```

Comment:
I thought we were removing the return value

Reply:
I rethought this a bit as I was editing it. I don’t think returning `True` if there are *any* successes will ever be useful, but I think there are scenarios where returning False in the event there are *any* failures could be useful.